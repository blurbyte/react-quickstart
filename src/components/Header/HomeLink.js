import { Link } from 'react-router-dom';
import styled from 'styled-components';

const HomeLink = styled(Link)`
  height: 4rem;
  border-right: 1px solid #908e89;
  display: inline-block;
  padding-top: 0.75rem;
  padding-right: 1.4rem;
  margin-right: 1.4rem;
`;

export default HomeLink;
