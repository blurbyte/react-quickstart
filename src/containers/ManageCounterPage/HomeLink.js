import NavLink from '../../components/NavLink';
import styled from 'styled-components';

const HomeLink = styled(NavLink)`
  margin-top: 4rem;
`;

export default HomeLink;
