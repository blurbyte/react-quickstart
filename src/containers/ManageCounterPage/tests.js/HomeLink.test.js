import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';

import HomeLink from '../HomeLink';

describe('<HomeLink />', () => {
  it('should render correctly', () => {
    // props init
    const props = {
      to: '/home'
    };
    // component init
    const component = renderer.create(
      <MemoryRouter>
        <HomeLink {...props} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
