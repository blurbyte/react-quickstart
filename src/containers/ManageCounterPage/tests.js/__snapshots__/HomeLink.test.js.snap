// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeLink /> should render correctly 1`] = `
.kQlxmu {
  font-family: 'OpenSans Light', Arial, Helvetica, sans-serif;
  color: #ff4326;
  font-size: 1.4rem;
  text-transform: uppercase;
  display: block;
  margin-right: 1.4rem;
  margin-top: 4rem;
}

.kQlxmu:hover {
  text-decoration: underline;
}

<a
  className="kQlxmu"
  href="/home"
  onClick={[Function]}
/>
`;
